lib/wallaroo/core/topology/router.pony

```diff
@@ -1352,61 +1720,51 @@ class val LocalPartitionRouter[In: Any val,
       in steps_to_migrate'.values()
     do
       router_registry.add_to_step_waiting_list(step_id)
-      step.send_state[Key](boundary, state_name', key)
-      router_registry.move_stateful_step_to_proxy[Key](step_id, step,
-        ProxyAddress(target_worker, step_id), key, state_name')
+      step.send_state(boundary, _state_name, key)
+      router_registry.move_stateful_step_to_proxy(step_id, step,
+        ProxyAddress(target_worker, step_id), key, _state_name)
       @printf[I32](
-        "^^Migrating step %s to outgoing boundary %s/%lx\n".cstring(),
-        step_id.string().cstring(), target_worker.cstring(), boundary)
+        "^^Migrating step %s for key %s to outgoing boundary %s/%lx\n"
+          .cstring(), step_id.string().cstring(), key.cstring(),
+          target_worker.cstring(), boundary)
     end
 
   fun blueprint(): PartitionRouterBlueprint =>
-    let partition_addresses = recover trn Map[Key, ProxyAddress] end
-    try
-      for (k, v) in _partition_routes.pairs() do
-        match v
-        | let s: Step =>
-          let pr = ProxyAddress(_worker_name, _step_ids(k)?)
-          partition_addresses(k) = pr
-        | let pr: ProxyRouter =>
-          partition_addresses(k) = pr.proxy_address()
-        end
-      end
-    else
-      Fail()
-    end
-
-    LocalPartitionRouterBlueprint[In, Key, S](_state_name, _step_ids,
-      consume partition_addresses, _partition_function)
+    LocalPartitionRouterBlueprint[In, S](_state_name, _step_ids,
+      _hash_partitions, _partition_function)
 
   fun distribution_digest(): Map[String, Array[String] val] val =>
```

Comment:
Do we need this anymore?  I think you replaced it with a different approach.

Reply:
I think we still need this.